ext/json/ext/parser/parser.rl

```diff
@@ -576,24 +631,8 @@ static char *JSON_parse_string(JSON_Parser *json, char *p, char *pe, VALUE *resu
 
     if (json->symbolize_names && json->parsing_name) {
       *result = rb_str_intern(*result);
```

Comment:
This also should be moved and use `rb_sym_intern`.

Reply:
Apparently `rb_sym_intern` was added to Ruby 2.3.0. I think I'll have to test for it.

Reply:
@nurse looks like `rb_sym_intern` isn't actually exposed to C extensions.